### PR TITLE
Move the code from sdks-java-fn-execution to sdks-java-core

### PR DESCRIPTION
**Pull request by kennknowles**
Part of fixing #30169

I have not completed debugging of the massive change at #29924 that would complete the issue. Instead, I am starting to break off pieces that are easier to complete, to shrink the final change.

This takes just the code from `sdks-java-fn-execution` and merges it into the core SDK. The dependencies look like this:

`sdks-java-core` <--  `sdks-java-fn-execution` <-- `runners-core-construction-java`

So the merge must necessarily flatten all of those. This smaller merge is easier to accomplish first. I suspect it may also be easier to move all the code from one module to another and separate the gradle build changes.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by kennknowles** [comment]
R: @chamikaramj

**Comment by kennknowles** [comment]
There are some tiny code changes that I guess were done as part of IntelliJ's "move" operation and subsequent `spotlessApply`. For now I'll leave them so CI can continue. They seem like fine to leave or I could do the manual work to undo them.

**Comment by kennknowles** [comment]
Finally java precommit green - some issue with windmill precommit

**Review by chamikaramj — COMMENTED** [review]
LGTM. Thanks.

File: sdks/java/fn-execution/build.gradle [hunk 1]
```diff
@@ -26,18 +26,9 @@ ext.summary = """Contains code shared across the Beam Java SDK Harness and Java
 the Beam Portability Framework."""
 
 dependencies {
-  implementation project(path: ":model:pipeline", configuration: "shadow")
-  implementation project(path: ":model:fn-execution", configuration: "shadow")
-  implementation project(path: ":sdks:java:core", configuration: "shadow")
-  implementation library.java.vendored_grpc_1_60_1
-  implementation library.java.vendored_guava_32_1_2_jre
```
[the remark this answers is not in the document]
Comment (kennknowles) [reply sdks/java/fn-execution/build.gradle:33]: Like maybe we can but I hit so many problems in the other PR that I tried to make the literally the smallest change I could. Let me try to push that as another commit on this PR...

[the remark this answers is not in the document]
Comment (chamikaramj) [reply sdks/java/fn-execution/build.gradle:33]: Sounds good. Thanks.

**Comment by kennknowles** [comment]
OK yea I hit problems - I think there are a LOT of modules that do not declare their dependencies. I think people are counting on IntelliJ to automatically edit the `build.gradle` and that is part of why they are so disorganized. And it doesn't always add what it should perhaps. Or who knows but either way, removing this module is its own challenge and I will get to it next.

**Pull request merged**